control_server/tests/admin/task_post_admin_test.py

```diff
@@ -0,0 +1,119 @@
+
+
+def randomize_ids() -> (str, str):
+    """
+    TODO: Add comment here
+    """
+    client_id = str(uuid.uuid4())
+    client_ip = str(uuid.uuid4())
+    return client_id, client_ip
+
+
+def set_client(client_id: str, client_ip: str):
+    """
+    Helper function for setting a client in the database
+    """
+    client_1 = {
+        "os_name": "2",
+        "os_version": "2",
+        "hostname": "2",
+        "host_user": "2",
+        "privileges": "2",
+    }
+    client_data = ClientData.load_from_dict(client_1, raise_error=True)
+    new_client = IdentifyingClientData(
+        client_data=client_data,
+        ip=client_ip
+    )
+    new_client.set_id(client_id)
+    controller.db.set_user(
+        user_id=client_id,
+        user=new_client,
+        overwrite=True
+    )
+
+
+def test_post_invalid_authorization(client):
+    """
+    Test the /admin/task post endpoint with missing authorization
+    :param client:
+    :return:
+    """
+    response = client.post(f"{get_prefix()}/admin/task", json={
+
+    })
+    assert response.status_code == 401, \
+        "Received a non-401 status code"
+
+
+def test_post_task_wrong_client_id(client):
+    """
+    Test the /admin/task post endpoint with wrong client id
+    :param client:
+    :return:
+    """
+    response_missing_client_id = client.post(f"{get_prefix()}/admin/task", headers={
+        "Authorization": controller.settings.admin_key
+    }, json={
+
+    })
+
+    response_wrong_client_id = client.post(f"{get_prefix()}/admin/task", headers={
+        "Authorization": controller.settings.admin_key
+    }, json={
+        "client_id": "1234",
+        "data": "test",
+        "name": "test",
+        "min_delay": "0",
+        "max_delay": "0"
+    })
+    assert response_missing_client_id.status_code == 400, \
+        "Received a non-400 status code"
+    assert response_wrong_client_id == 404, \
+        "Received a non-404 status code"
+
+
+def test_post_task_missing_task(client):
+    """
+    Test the /admin/task post endpoint with missing task
+    :param client:
+    :return:
+    """
+    response_missing_task = client.post(f"{get_prefix()}/admin/task", headers={
+        "Authorization": controller.settings.admin_key
+    }, json={
+        "client_id": "1234"
+    })
+    assert response_missing_task.status_code == 400, \
+        "Received a non-404 status code"
+
+
+def test_post_task(client):
+    """
+    Test the /admin/task post endpoint
+    :param client:
+    :return:
+    """
+    # Set client in DB
+    client_id, client_ip = randomize_ids()
+    set_client(client_id=client_id, client_ip=client_ip)
+
+    response = client.post(f"{get_prefix()}/admin/task", headers={
+        "Authorization": controller.settings.admin_key
+    }, json={
+        "client_id": client_id,
+        "data": "test",
+        "name": "test",
+        "min_delay": "0",
+        "max_delay": "0"
+    })
+
+    assert response.status_code == 200, \
```

Comment:
Would be nice to try to get the posted task from the database here as well, and assert that its the same as the task posted.

Reply:
Added assertions for check that the posted task is in DB.